test/test_heuristics.rb

```diff
@@ -439,20 +439,31 @@ def test_ice_by_heuristics
   end
 
   def test_inc_by_heuristics
+    ambiguous_pascal = [
+      "#{samples_path}/Pascal/cwindirs.pp",
+      "#{samples_path}/Pascal/image_url.inc",
+      "#{samples_path}/Pascal/large.pp",
+      "#{samples_path}/Pascal/libc.inc",
+      "#{samples_path}/Pascal/program.dpr",
+      "#{samples_path}/Pascal/tw27294.pp",
+      "#{samples_path}/Pascal/large.pp",
+      "#{samples_path}/Pascal/vmops_impl.inc"
+    ]
     assert_heuristics({
       "Motorola 68K Assembly" => all_fixtures("Motorola 68K Assembly", "*.inc"),
       "NASL" => all_fixtures("NASL", "*.inc"),
+      "Pascal" => all_fixtures("Pascal", "*") - ambiguous_pascal,
```

Comment:
This doesn't feel right to me. We're pulling in all Pascal samples, including 3 `.inc` files, only to then remove _all_ but one: `samples/Pascal/GraphicConfiguration.inc`.

This suggests to me the heuristic has scope for improvement. 

Could the heuristic be improved to catch all of the samples we've already got? Maybe matching on the use of `begin` and `end`?

Reply:
In case that helps, `begin` and `end` would not always be present in .inc files, only through comments or in some FreePascal source, the are however almost present in .pas/.pp/.dpr
`$define` can be present in all Pascal file types.

Reply:
Right. `$define` was already covered. I added an additional pattern that covers all Pascal samples. Also increased the number of samples tested for other languages (not just those originally with `.inc`) to increase coverage of this heuristic.

All of them are fine except C++, which is sometimes classified as NASL, but I'll address that in a separate PR.